projects/core/components/textfield/textfield.component.ts

```diff
@@ -181,7 +181,9 @@ export class TuiTextfieldComponent<T> implements TuiDataListHost<T>, AfterConten
 
     public handleOption(option: T): void {
         this.accessor?.setValue(option);
-        this.open.set(false);
+        // Ensure all external (click) handlers are called before closing dropdown
+        // <button tuiOption (click)="..." />
+        setTimeout(() => this.open.set(false));
```

Comment:
what do you think? 

```ts
delay(0).pipe(takeUntilDestroyed(this.destroyRef)).subscribe(
 () => this.open.set(false)
);
```

Reply:
We decided it is not necessary for such cases